pages/Documents/ConversationalAI/Third-PartyBots/GoogleDialogflowCX/advanced-features.md

```diff
@@ -0,0 +1,291 @@
+
+```json
+{
+  "metadata": [
+    {
+      "id": "1234",
+      "type": "ExternalId"
+    }
+  ],
+  "structuredContent": {
+    "type": "vertical",
+    "elements": [
+      {
+        "type": "button",
+        "click": {
+          "actions": [
+            {
+              "text": "Recommend me a movie, please",
+              "type": "publishText"
+            }
+          ]
+        },
+        "title": "Recommend a movie"
+      }
+    ]
+  },
+  "encodedMetadata": "ewoic29tZUluZm8iOiAiSSB3YXMgZW5jb2RlZCIKfQ=="
+}
+```
+
+<br />
+
+<img class="fancyimage" style="width:600px" src="img/dialogflowcx/dialogflow_encodedmetadata_structuredcontent.png">
+Figure 3.3
+
+### Sending Pause/Delay Message
+
+It is possible to send an event of type "delay" before regular content events and actions. This specifies the time the bot will wait before displaying the next message. The delay message can be added via the Custom Payload response in intent definition (as shown in Figure 3.4). There are two properties, `delay` and `typing`, which are a part of the Custom Payload response.
+
+<ul>
+  <li> <b>delay</b>: This is the number of seconds the bot will wait. These are expected to be only whole numbers for example for one second delay you will write 1 as a value</li>
+  <li><b>typing</b>: This property will enable/disable the typing indicator while delay is happening. It is optional; if not provided then the value will be considered as true.</li>
+</ul>
+
+<br />
+
+Setting a delay in between multiple messages is possible and an example of such a case (Message - Delay - Message) can be seen in Figure 3.4.
+
+<img class="fancyimage" style="width:600px" src="img/dialogflowcx/dialogflow_message_delay_message.png">
+Figure 3.4 An example of Message - Delay - Message  configuration in the Dialogflow console's intent editor
+
+It is possible to send only a single delay response. The example payload of such response is below:
+
+```json
+{
+  "delay": 8,
+  "typing": false
+}
+```
+
+<img class="fancyimage" style="width:800px" src="img/dialogflowcx/delay_response_custom_payload.png">
+Figure 3.5 showing the Custom Markup message for delay message
+
+**Note:** using the delay as a single/sole response from the bot to the consumer, is effectively a ‘no response’ action. Using this allows the bot to receive a consumer message without responding to the consumer.
+
+### Sending Private Text Message
+
+It is possible to send a private text message from the Live Engage (LE-UI) via agent workspace. This feature can now be used via the Third-Party bots as well. This will allow Brands to define private message text within the conversational flow of the bot. These messages are published into the conversation for other Agent/Manger participants. This enables Brands to customize messages giving more insight, summarizing actions taken by the bot, or also advising on next actions the handover agent should take.
```

Comment:
```suggestion
It is possible to send a private text message from the Conversational Cloud to other agents on the conversation. This feature can now be used via the Third-Party Bots as well. This will allow Brands to define private message text within the conversational flow of the bot. These messages are published into the conversation for other Agent/Manger participants. This enables Brands to customize messages giving more insight, summarizing actions taken by the bot, or also advising on next actions the handover agent should take.
```